File: [프로그래머스] 연습문제_추억 점수.py
def solution(name, yearning, photo):
    answer = []
    dic = {name:yearning for name, yearning in zip(name, yearning)}
    for i in range(len(photo)):
        answer.append(0)
        for j in photo[i]:
            if j in dic.keys():
                answer[i] += dic[j]
    return answer

File: [프로그래머스] 연습문제_추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    let dic = {}
    for(let i = 0; i < name.length; i++){
        dic[name[i]] = yearning[i];
    }
    for(let i = 0; i < photo.length; i++){
        answer.push(0);
        for(let j of photo[i]){
            if(name.includes(j)){
                answer[i] += dic[j];
            }
        }
    }
    return answer;
}